Transfering DB to RPC (Ticket #125125311), Transfering Dashboard to RPC (Ticket #125119911), Transfering Editor to RPC(Ticket #125119661), Transfering 2D-model to RPC(Ticket #125119771), Adding Travis CI (Ticket #126298269)

TanVD opened the pull request.

TanVD committed:
Ticket #125119911
1.Added thrift files for robotrestcontroller
2.Main application config changed. Context now transparent.
3.New rpc controller for robotrest created. Old deleted.

TanVD committed:
Ticket #125119911
1.Unused files deleted
2.One user will be created automaticly
3.Minor improvements

artbez committed:
Add bash file for auto compilation .thrift

artbez committed:
add UserDbServer with dog-nail

artbez committed:
Get rid of userService in UserDbServiceHandler

artbez committed:
Get rid of userService in whole project

TanVD committed:
Ticket #125125311
1.Fixed problem with UserService (findByUsername not working)
2. Added -p in thriftGen.sh

artbez committed:
Tickets #125119911 and #125125311
1. Add thrift structures TUserRole and TRobot
2. Add converter between User and TUser
3. Fix bug with creating robots in dashboard

saimonsaret committed:
Convert editor ajax requests to RPC-requests (partially) (ticket #125119661)

Save diagram request is now RPC-request with a Thrift use

saimonsaret committed:
Convert update request + bugfix (ticket #125119661)

artbez committed:
Tickets #125119911 and #125125311
1. Add database service for robots
2. Link this service with dashboard_Rest_servic

artbez committed:
Some cosmetic changes

TanVD committed:
Ticket #126298269
1. Travis CI .yml file added in project
2. Travis CI integration added in slack

TanVD committed:
Ticket #126298269 (Travis CI)
1. In Travis CI .yml file thrift installing instructions added

TanVD committed:
Ticket #126298269 (Travis CI)
1. In Travis CI .yml file -y added for all installing instructions

TanVD committed:
Ticket #126298269 (Travis CI)
1. In Travis CI .yml file upgrade instruction deleted

TanVD committed:
Ticket #126298269 (Travis CI)
1. In Travis CI .yml file trusty dist now used

TanVD committed:
Ticket #126298269 (Travis CI)
1. In Travis CI .yml file trusty dist now used correctly (mistake corrected)

TanVD committed:
Ticket #126298269 (Travis CI)
1. Changed order. Version of thrift 0.9.0.3 didn't understand old one.

TanVD committed:
Ticket #126298269 (Travis CI)
1. Tomcat test added

TanVD committed:
Ticket #126298269 (Travis CI)
1.Tomcat test fixed.

saimonsaret committed:
Comlplete server side RPC-transfer

artbez committed:
Merge Editor_RPC

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/robots/service/client/RobotServiceImpl.java:
```diff
@@ -0,0 +1,59 @@
+    @Autowired
+    private UserService userService;
+
+    TTransport transport;
```
давайте всё же писать модификатор доступа в явном виде везде. и располагать секции по уменьшению области видимости

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/robots/service/server/RobotDbServiceHandler.java:
```diff
@@ -0,0 +1,69 @@
+        try {
+            user = userService.findByUserName(tRobot.getUsername());
+        } catch (TException e) {
+            e.printStackTrace();
```
а чего мы их игнорируем везде?

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/robots/service/server/RobotDbServiceHandler.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    private boolean userRobotExists(User user, String name) {
+        for (Robot robot : user.getRobots()) {
```
```
return user.getRobots().stream().anyMatch(r -> r.getName().equals(name));
```

давайте не будем писать говнокод в новый репозиторий :)
новая жизнь с чистого листа!

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/robots/service/server/RobotDbServiceHandler.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    public static Robot convertToRobot(TRobot tRobot, User user) {
+
```
строчка лишняя пустая

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/robots/service/server/RobotDbServiceHandler.java:
```diff
@@ -0,0 +1,69 @@
+    public static TRobot convertFromRobot(Robot robot) {
+        String username = robot.getOwner().getUsername();
+        TRobot tRobot = new TRobot(robot.getName(), robot.getSsid(), username);
+        int a = 0;
```
мощно.

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+@Service("UserService")
+public class UserServiceImpl implements UserService {
+
+    TTransport transport;
```
видимость бы добавить. и вниз перетащить

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+}
+
+class ConverterFromT {
+
```
лишняя строчка

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+    static Set<UserRole> castFromTRole(Set<TUserRole> tUserRoles, User user) {
+        Set<UserRole> userRoles = new HashSet<>();
+        for (TUserRole tur : tUserRoles) {
+
```
ещё одна

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+        user.setRobots(castFromTRobot(tUser.getRobots(), user));
+        return user;
+    }
+
```
больше пустых строк богу пустых строк!

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -0,0 +1,129 @@
+
+}
+
+class ConverterToT {
```
не очень хорошее имя класса

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServer.java:
```diff
@@ -0,0 +1,43 @@
+import org.springframework.context.support.AbstractApplicationContext;
+
+/**
+ * Created by artemiibezguzikov on 11.07.16.
```
скажем решительное НЕТ мезкому духу сатанинской гордыни и бесовского тщеславия!

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServiceHandler.java:
```diff
@@ -0,0 +1,109 @@
+}
+
+class ConverterFromT {
+
```
лишняя строчка

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServiceHandler.java:
```diff
@@ -0,0 +1,109 @@
+    }
+}
+
+class ConverterFromT {
```
ну и это я бы тоже переименовал по аналогии

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServiceHandler.java:
```diff
@@ -0,0 +1,109 @@
+    static Set<UserRole> castFromTRole(Set<TUserRole> tUserRoles, User user) {
+        Set<UserRole> userRoles = new HashSet<>();
+        for (TUserRole tur : tUserRoles) {
+
```
пустая строчка

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServiceHandler.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    static Set<Robot> castFromTRobot(Set<TRobot> tRobots, User user) {
+        Set<Robot> robots = new HashSet<>();
```
тут должно быть что-то типа такого:

```
return tRobots.stream().map(trob -> new Robot(trob.getId(), trob.getName(), trob.getSsid(), user)).collect(Collectors.toSet());
```

ну и во всех аналогичных случаях тоже. нам за строчки кода не платят :)

saimonsaret committed:
Convert editor ajax requests to RPC-requests (completely) (ticket #125119661)

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServiceHandler.java:
```diff
@@ -0,0 +1,109 @@
+
+}
+
+class ConverterToT {
```
а это всё (или очень близкое) уже было в UserServiceImpl. мы не хотим этот код как-то переиспользовать?

TanVD replied:
Тот и этот код будут на разных сервисах. Можно выделить в пакет и при разбиении раскопировать:)

jzuken replied:
можно так, но не копипастой же

jzuken commented on Robots_diagram/src/main/webapp/WEB-INF/views/dashboard/index.jsp:
```diff
@@ -18,115 +20,106 @@
-                    }
-                });
-            });
+//            $('[name="sendDiagram"]').click(function (event) {
```
а чего он закомментирован? надо или оставить коммент на эту тему, или убрать, если оно не нужно

artbez committed:
Prepare for adding DiagramDbService

artbez committed:
Merge some improvements from Editor_RPC

artbez committed:
Change names of thrift datastructures

artbez committed:
Add diagram db service
1. Add property optional for some fields
2. Fix EditorConverter
3. Get rid of DiagramRequest
4. Rewrite DiagramService to RPC

artbez committed:
Add idea files

artbez committed:
Add .iml files

artbez committed:
Fix bug with robotService

TanVD committed:
Ticket #126298269 (Travis CI)
1.File renamed to fit thriftGen script

TanVD committed:
Ticket #126298269 (Travis CI)
    1.Travis yml file changed to load 0.9.3 version from site

TanVD committed:
Ticket #126298269 (Travis CI)
1.Fixed order of operations in yml. Comments added

TanVD committed:
Ticket #126298269 (Travis CI)
1. Some languages were disabled in thrift configuration

jzuken commented on Robots_diagram/thriftGen.sh:
```diff
@@ -0,0 +1,48 @@
+#client side
+editorClientDir="src/main/webapp/resources/thrift/editor"
+mkdir -p "$editorClientDir"
+thrift -gen js:ts -out "$editorClientDir" "$editorThriftDir/service/EditorService.thrift"
```
файл называется editorService.thrift, линукс неодобряэ

jzuken replied:
и этому файлу надо руками chmod +x делать

jzuken commented on Robots_diagram/thriftGen.sh:
```diff
@@ -0,0 +1,48 @@
+
+#Authorization service
+authThriftDir="src/main/java/com/qreal/robots/components/authorization/thrift"
+mkdir -p "$authThriftDir/gen"
```
а если такая директория есть, оно просто свалится тут. давайте-ка лучше мавеном это всё делать

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1. Code of classes formatted according to styleguide
2. Redundant code deleted from interfaces

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Logging added
2.Minor improvements all over the code

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.thrift.d.ts renamed as Thrift.d.ts (matter for Linux)

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/common/exception_controller/ExceptionHandlerController.java:
```diff
@@ -8,13 +11,17 @@
 
     @ExceptionHandler(Exception.class)
     public ModelAndView handleException(Exception e) {
+        logger.warn("Someone met exception");
```
ещё было бы неплохо само e добавить в лог

TanVD replied:
Исправлю. В остальныех местах я везде передавал исключение последним аргументом в log4j, тут что-то пропустил :)

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/authorization/controller/AuthController.java:
```diff
@@ -51,30 +56,28 @@ public ModelAndView register(@RequestParam(value = "username") String username,
 
-            userService.save(user);
+        if (userService.isUserExist(username)) {
+            return registerError(model, String.format("User with %s name is already exist", username));
```
а почему не тупо конкатенация? я бы заменил, читаемость лучше будет

TanVD replied:
Это ещё код из репозитория Захарова, просто не менял. Ок

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/authorization/controller/AuthController.java:
```diff
@@ -51,30 +56,28 @@ public ModelAndView register(@RequestParam(value = "username") String username,
-            e.printStackTrace();
+
+        if (!password.equals(password2)) {
+            return registerError(model, String.format("Passwords are not equals"));
```
тут тоже можно было бы записать что-то в лог

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/dashboard/controller/MainController.java:
```diff
@@ -23,27 +24,20 @@
+
     public static final String HOST_NAME = "127.0.0.1";
 
     public static final int PORT = 9002;
```
что-то этот номер из шаблона выбивается. остальные вроде как 909x

TanVD replied:
А он относится к эфемерному robot routing service разработки вроде агеева... сюда хоть 666 ставь, оно всё равно никогда не вызывается :)
Можно в принципе пока убрать совсем его.

jzuken replied:
я голосую за убрать всё, что не используется. если кому-то когда-то этот код понадобится, возродят

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/client/UserServiceImpl.java:
```diff
@@ -67,7 +98,8 @@ public boolean isUserExist(String username) throws TException {
             userRoles.add(new UserRole(tur.getUserRoleId(), user, tur.getRole()));
-        };
+        }
+        ;
```
wat?

jzuken commented on Robots_diagram/src/main/java/com/qreal/robots/components/database/users/service/server/UserDbServer.java:
```diff
@@ -1,44 +1,48 @@
 
-    public static void simple(UserDbService.Processor processor) {
+    public static void runTServer(UserDbService.Processor processor) {
+        int port = 9090;
```
а получится все порты в один конфиг вынести?

TanVD replied:
Сейчас да. В дальнейшем не совсем понятно как это делать, если развёртываться на разных машинах будет.

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Cloned classes deleted
2.Thrift files structure changed
3.Converters changed
4.Contracts for DB added
5.Minor improvements

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1. Unnecessary files deleted
2. Minor improvements

artbez committed:
Make Robots identified by id (not robotName)

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Docs added
2.Checkstyle added
3.Minor improvements all over the code according checkstyle inspection

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Merged latest commit by artbez (fixing issue with robot's db interface)

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Checkstyle fixes

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
    1.Checkstyle fixes

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Call tomcat test now waits for 20 min
2.Travis now firstly runs mvn clean install and only after runs tomcat7:run-war

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Unused files and projects deleted
2.Log file added

TanVD committed:
Ticket #126856635(Refactoring of major pull request)
1.Fixes for codacy
2.Packages now should be in lower case

jzuken committed:
updated pom files. renamed web service into robots-editor
